dbt_style_guide.md

```diff
@@ -1,237 +1,522 @@
+    - base_stripe__invoices.sql
+    - stg_stripe__customers.sql
+    - stg_salesforce__customers.sql
+    - int_customers__unioned.sql
+    - fct_orders.sql
 
-* Schema, table and column names should be in `snake_case`.
-* Use names based on the _business_ terminology, rather than the source terminology.
-* Each model should have a primary key.
-* The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
-* For base/staging models, fields should be ordered in categories, where identifiers are first and timestamps are at the end.
-* Timestamp columns should be named `<event>_at`, e.g. `created_at`, and should be in UTC. If a different timezone is being used, this should be indicated with a suffix, e.g `created_at_pt`.
-* Booleans should be prefixed with `is_` or `has_`.
-* Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
-* Avoid reserved words as column names
-* Consistency is key! Use the same field names across models where possible, e.g. a key to the `customers` table should be named `customer_id` rather than `user_id`.
+## Naming Conventions
 
-## CTEs
+- Schema, table and column names should be in `snake_case`.
 
-For more information about why we use so many CTEs, check out [this discourse post](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091).
+- Limit use of abbreviations that are related to domain knowledge. An onboarding
+  employee will understand `current_order_status` better than `current_os`.
 
-- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file
-- Where performance permits, CTEs should perform a single, logical unit of work.
-- CTE names should be as verbose as needed to convey what they do
-- CTEs with confusing or noteable logic should be commented
-- CTEs that are duplicated across models should be pulled out into their own models
-- create a `final` or similar CTE that you select from as your last line of code. This makes it easier to debug code within a model (without having to comment out code!)
-- CTEs should be formatted like this:
+- Use names based on the _business_ terminology, rather than the source terminology.
+
+- Each model should have a primary key that can identify the unique row.
 
-``` sql
-with
+- The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
 
-events as (
+- If a surrogate key is created, it should be named `<object>_sk`.
 
-    ...
+- For `base` or `staging` models, columns should be ordered in categories, where identifiers are first and date/time fields are at the end.  
+  Example:
+  ```sql
+  transformed as (
+      select

```

Comment:
I'm sure there were debates on the newline aspect but I actually prefer newlines.

They give more space for the eyes.

Reply:
@ericalouie Thanks for commenting this one! It's expected that we'll rehash some debates now that ProServ has aligned and we've opened our thoughts to everyone else. Battle of the teams, commence! ⚔️ 

I'm gonna keep this open so that peeps can 👍 or 👎 =)

Reply:
I personally don't feel strongly about adding newlines after the CTE name or not.

Reply:
For the time being I'm leaving this as is!